react/ProductHighlights.tsx

```diff
@@ -36,7 +36,8 @@ interface ProductHighlightContextProviderProps {
 
 function createFilterHighlight(filter: Filter) {
   return function filterHighlight(highlight: Highlight) {
-    const hasHighlight = filter.highlightNames.includes(highlight.name)
+    const namesToFilter = filter.highlightNames ?? [] 
+    const hasHighlight = namesToFilter.includes(highlight.name)
```

Comment:
```suggestion
    const hasHighlight = filter.highlightNames?.includes(highlight.name)
```

I would suggest only an [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) here